packages/as-types/assembly/__tests__/serializer.spec.ts

```diff
@@ -20,10 +20,15 @@ import {
 } from '../serialization';
 
 describe('Serialization tests', () => {
-  it('ser/deser string', () => {
+  it('ser/deser Utf-16 string', () => {
```

Comment:
Are you sure it's UTF-16 encoded ? The file is not encoded in UTF8 ?

To be sure I would have used [String.fromCharCode()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCharCode)

Reply:
the "🙂" emoji is [U+1F642](https://unicode.org/emoji/charts/full-emoji-list.html#1f642) unicode. which gives f0 9f 99 82 in utf8, so 4 bytes.
UTF-16 is variable length with code points encoded with one or two 16-bit code units
So the 🙂 emoji is a charactere encoded on 32 bits. and if we see its as an emoji, its mean its utf16 and your editor support it for display

Reply:
but maybe i don't understand it properly....